control/controlexecute/control_run.go

```diff
@@ -171,22 +170,25 @@ func (r *ControlRun) Execute(ctx context.Context, client db_common.Client) {
 	r.Lifecycle.Add("queued_for_session")
 	dbSession, err := client.AcquireSession(ctx)
 	if err != nil {
+		if utils.IsCancelledError(err) {
```

Comment:
Because we have extra `Execute`s running, `client.AcquireSession` may fail with a `Context.Cancelled`

Reply:
maybe change this to 
```
if !utils.IsCancelledError(err){
    err = fmt.Errorf("error acquiring database connection, %s", err.Error())
}
r.SetError(err)
return
```